### DISCUSSION TITLE
Hidden entity filter card uses space

### DISCUSSION THREAD

**Discussion by cgtobi** [home-assistant/frontend#11192] [Other feature requests]
Entity filter cards apparently use space even when they are hidden (if they are empty). Is that expected behaviour? 

![lhiszid](https://user-images.githubusercontent.com/856782/43086380-80e3c3da-8e9d-11e8-8b48-1b8d1ae2970a.png)

**Comment by balloob** [comment]
Hmmm, that's an unfortunate side effect. The layout engine doesn't know a card is going to decide to show itself yes or no. So it will divide the screen in columns based on that a card config _will_ produce a card.

**Comment by jeradM** [comment]
@balloob what do you think about allowing cards to define an optional `willRender` method that can be called while building the columns? If that method exists and returns `false`, we skip that card. Then if something changes in a card that will cause it to render, it can fire `rebuild-view`. The downside is that when this happens the whole view is rebuilt, but it would solve the current issue

**Comment by Misiu** [comment]
@jeradM I think this isn't the best solution.
I'd prefer to have a card with the text `No valid entities` or similar instead of hiding that card completely.
Imagine you have a couple of filter cards and they change quite often. Your UI will be a total mess.
I think that rebuilding the whole view should be avoided because it is time and resource consuming.
Please correct me if I'm wrong - I'm still new to HA, but I have a C# (WinForms) background - there is one rule when invalidating (repainting) components - smallest region possible. I think the same applies here.

**Comment by kg333** [comment]
The same effect occurs with conditional cards.  I have a number of these for alerts of important or unusual states (security alarm engaged, cameras in full recording mode, part of home network not responding, etc), but they cause my main cards to push to the right even if the conditionals are currently hidden:
![image](https://user-images.githubusercontent.com/4220833/77793073-d8c10380-703f-11ea-96f6-0e8252f51589.png)

![image](https://user-images.githubusercontent.com/4220833/77792881-87b10f80-703f-11ea-8e2c-5d51e833fdfa.png)

**Comment by KTibow** [comment]
How about setting `cardHeight` to 0?

**Comment by kg333** [comment]
@KTibow, somewhat interesting result by setting the conditional Markdown cards with `card_size: 0` - they no longer shift other cards to the right, but also don't distribute themselves horizontally when they are visible.  This also doesn't work for conditional cards that don't have that property (Alarm Panel in my case).
![image](https://user-images.githubusercontent.com/4220833/92060254-4137ad80-ed61-11ea-9900-5cfc956d4413.png)

Are there any other ways to force `cardHeight`?

**Comment by KTibow** [comment]
I think you might be able to do it with `card-mod`, `mod-card`, or `layout-card`.

**Comment by kg333** [comment]
No impact with lovelace-card-mod style:
```
style: |
  ha-card {
    height: 0%;
  }
```

Using `custom:mod-card` with `report_size: 0` results in same behavior as setting conditional Markdown cards with `card_size: 0`.  Cards all end up vertically stacked.

**Comment by KTibow** [comment]
Did you apply the size to the conditional card or the markdown card?

**Comment by kg333** [comment]
To the conditional card itself, rather than the contained cards.
